src/main/java/com/zaxxer/hikari/util/JavassistProxyFactory.java

```diff
@@ -222,13 +224,13 @@ private static boolean isDefaultMethod(Class<?> intf, CtClass intfCt, CtMethod i
          paramTypes.add(toJavaClass(pt));
       }
 
-      return intf.getDeclaredMethod(intfMethod.getName(), paramTypes.toArray(new Class[paramTypes.size()])).toString().contains("default ");
+      return intf.getDeclaredMethod(intfMethod.getName(), paramTypes.toArray(new Class[0])).toString().contains("default ");
    }
 
    private static Set<Class<?>> getAllInterfaces(Class<?> clazz)
    {
       Set<Class<?>> interfaces = new HashSet<>();
```

Comment:
Instead of all the comparator fluff above couldn't we just use a linked hash map here?

Reply:
Yes, this is definitely a better solution.